plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1087,29 +1087,133 @@ public void testCaseInsensitivePredicate()
+                    .isNotFullyPushedDown(FilterNode.class);
             return;
         }
-        assertThat(query("SELECT shippriority FROM orders WHERE shippriority % 4 = 0")).isFullyPushedDown();
+        // negate
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE -(nationkey) = -7"))
+                .matches("VALUES (BIGINT '7', CAST('GERMANY' AS varchar(25)), BIGINT '3')")
+                .isFullyPushedDown();
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE -(nationkey) = 0"))
+                .matches("VALUES (BIGINT '0', CAST('ALGERIA' AS varchar(25)), BIGINT '0')")
+                .isFullyPushedDown();
 
-        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % nationkey = 2"))
-                .isFullyPushedDown()
-                .matches("VALUES (BIGINT '3', CAST('CANADA' AS varchar(25)), BIGINT '1')");
+        // additive identity
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey + 0 = nationkey"))
+                .matches("SELECT nationkey, name, regionkey FROM nation")
+                .isFullyPushedDown();
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey - 0 = nationkey"))
+                .matches("SELECT nationkey, name, regionkey FROM nation")
+                .isFullyPushedDown();
+        // additive inverse
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey + (-nationkey) = 0"))
+                .matches("SELECT nationkey, name, regionkey FROM nation")
+                .isFullyPushedDown();
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey - nationkey = 0"))
+                .matches("SELECT nationkey, name, regionkey FROM nation")
+                .isFullyPushedDown();
 
-        // some databases calculate remainder instead of modulus when one of the values is negative
-        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % -nationkey = 2"))
-                .isFullyPushedDown()
-                .matches("VALUES (BIGINT '3', CAST('CANADA' AS varchar(25)), BIGINT '1')");
+        // constant
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey + 1 = 7"))
+                .matches("VALUES (BIGINT '6', CAST('FRANCE' AS varchar(25)), BIGINT '3')")
+                .isFullyPushedDown();
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey - 1 = 7"))
+                .matches("VALUES (BIGINT '8', CAST('INDIA' AS varchar(25)), BIGINT '2')")
+                .isFullyPushedDown();
+
+        // column reference
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey + regionkey = 26"))
```

Comment:
what is "a column reference"?

Reply:
changed to `// addition and subtraction of columns` and `// addition and subtraction of columns ANDed with another predicate`.